renderdoc/driver/metal/metal_helpers_bridge.mm

```diff
@@ -0,0 +1,41 @@
+/******************************************************************************
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2022 Baldur Karlsson
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ ******************************************************************************/
+
+#include "metal_helpers_bridge.h"
+#import <AppKit/AppKit.h>
+#import <Foundation/NSStream.h>
+
+void ObjC::Get_defaultLibraryData(void *&pData, uint32_t &bytesCount)
```

Comment:
I thought that metal-cpp meant there wasn't a need to jump into Obj-C like this? is there no metal-cpp equivalent of this call?

Reply:
Unfortunately, this isn't covered by metal-cpp. 
This isn't Metal API, it is Foundation API which is Objective C only.
This method is similar to things like
`void getMetalLayerSize(void *layerHandle, int &width, int &height)` which is in `vk_apple_helpers.mm `or the helper methods in `apple_helpers.mm`

Could move this helper method (and there will be a few more in future PRs) into one of the existing objective C files ie. `apple_helpers.mm` ?

Reply:
Oh I see. This is probably fine then - I'm presuming obj-c won't support the `bytebuf` type which would be preferred for use here instead of a raw `void*`/`uint32`? If not, then this is fine as-is.

Reply:
I will investigate bytebuf : it should be possible to construct a bytebuf from the obj-c data in this method

JAKE :TODO:

Reply:
That looks a lot cleaner

3a3c60864d1f66100080999ace37daf29d55459b